src/core/geocms/geonode/qgsgeonoderequest.h

```diff
@@ -93,7 +150,7 @@ class CORE_EXPORT QgsGeoNodeRequest : public QObject
      *
      * If \a forceRefresh is false, then cached copies of the request may be reused.
      */
-    QgsGeoNodeRequest( const QString &baseUrl, bool forceRefresh, QObject *parent = nullptr );
+    QgsGeoNodeRequest( const QString &baseUrl, bool forceRefresh, QgsGeoNodeAuthorization *geonodeAuth = nullptr, QObject *parent = nullptr );
```

Comment:
@ismailsunni `geonodeAuth` is a nullptr here, so your code will crash when you try to set the login/password in your constructor ;-)

Reply:
Ah, thanks @gustry. Now another problem, I couldn't login with basic auth. After discussion with @lucernae either he will make one or I will do the hacky way.